openstack/baremetal/v1/nodes/requests.go

```diff
@@ -247,6 +247,9 @@ type CreateOpts struct {
 
 	// A string or UUID of the tenant who owns the baremetal node.
 	Owner string `json:"owner,omitempty"`
+
+	// Static network configuration to use during deployment and cleaning.
+	NetworkData map[string]interface{} `json:"network_data,omitempty"`
```

Comment:
To confirm: this is just an arbitrary/generic map/dict/hash that will accept any data?

https://github.com/openstack/ironic/blob/462969c68f2978b4b250ba99ef491ef6858437fe/ironic/objects/node.py#L168

Normally if a field was added from a microversion, it should be a pointer so that the field is omitted from the boddy if it isn't set. But it looks like Ironic is accounting for having this field set when the microversion is less than 1.35: https://github.com/openstack/ironic/blob/462969c68f2978b4b250ba99ef491ef6858437fe/ironic/objects/node.py#L554

Because of this, I'm okay with leaving it as `map[string]interface{}`, but if you wanted to change it to `*map[string]interface{}`, go for it.

Reply:
> If you're OK with it I think I'd prefer to leave it, because there's no useful distinction between the field being empty vs. not set.

In this case, I agree. Usually it's because the API will return an error for an invalid field when the microversion isn't set correctly.